### ISSUE TITLE
Do we want to add comments for ESLint warnings?

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-code-manager#174]
When running `ESLint` on the current code base, we get the following warnings:

```
/addons-code-manager/src/test-helpers.spec.tsx
  14:33  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/addons-code-manager/src/test-helpers.tsx
  170:41  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  171:48  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 3 problems (0 errors, 3 warnings)
```

which all have to do with `ShallowUntilTarget`. I think our use of `any` in those cases makes sense, but it's a bit annoying to see these warnings every time we run `ESLint`.

How do folks feel about adding code comments to the files to ignore those specific warnings in those cases?

**Comment by willdurand** [comment]
I think we should make this rule throw an error instead of a warning. We can still disable the rule if needed. The patch for this issue will determine whether each current warning needs to be fixed or ignored.

**Comment by kumar303** [comment]
> I think we should make this rule throw an error instead of a warning. We can still disable the rule if needed.

I agree. This will make the rule much more helpful.

I think warnings are dangerous because they are easy to ignore and they clutter up the terminal which makes solving errors more difficult.

**Comment by willdurand** [comment]
This has been fixed in #238.
